### PR TITLE
New travis matrix

### PR DESCRIPTION
**Pull request by lasofivec**
Created a matrix for travis testing: 

- Now on mac, linux (trusty and xenial)
- Deployment only on master branch
- Deployment of wheel package if on mac
- Added skip_existing so that all build pass always
- Added possibility to test deployment if push to `deploy-test` branch.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,69 +1,120 @@
...
+- echo "BEFORE DEPLOY START........"
 - ls $START
 - cd $START
+- conda config --set anaconda_upload no
+- conda install anaconda-client conda-build
+- conda build conda_recipe
+- export PKG_REAL=$(conda build . --output | tail -1)
+- echo "BEFORE DEPLOY END.........."
 deploy:
-  - provider: script
```
Comment (Didou09) [inline .travis.yml]: If this is not done anymore, where is the package uploaded to anaconda.org ?

Comment (lasofivec) [reply .travis.yml]: Forgot to add it ! Sorry and thanks for catching it !

Comment (lasofivec) [reply .travis.yml]: (corrected in last commit)

**Comment by lasofivec** [comment]
> * There is more explicit error message for the volume computation error, which is good, but has the error itself been corrected somewhere ? (another branch ?)

Issue #314 was closed by PR #323

File: .travis.yml [hunk 2]
```diff
@@ -1,69 +1,127 @@
...
+      branch: master
+    password:
+      secure: JNEDTDJVx/2fXNfHntNQ99iDRNuQ4uB3y+DBWVIBycCT95+UCb36YPtKzmruEk/UUS29Xgq4IYCGdfCSWE9smKqG8tV1PcHiw705m+AzcpKy77YtzbVECFBxqY4W36O2pHrkwEUzP/7acjFwNsnUFzArqEzsBJ+KdLaa4OPHJXCh30GA0GyqlrXYbBKG+DA9hX5vtsGo4C6w9noALYF3fS7pKPiI6ipKFnAlzGgHQ7Ke0uQME8N3IAFhmh+Z5xMtIIDWxlnqv+KszdG4DIaGV/W6NIJNAbRhzkqUd+Chu6LoPAd/XkHDTeirR/MBkNUc5UcRJxRnP9rUTRo1gCO/buTYuNRgFkMvqhV5a033+x9edWgtUiKNJIMPLXOxe0RJvc5GWji+Co77HtHxRmGRM2rnYqWMtZeYZlFbUdvHu/8jf0d6I8jyUgAoJYdlMA2u/ipENP3S6by4epE9qycUPXiIVh6r3DZbf3vPTMFvTZYAjBrA0NOzihv1xgcXwemmNUFOQSpe0io4UcFxtS9lLMo+30UMQjCHSnbEVM3zSlZmbMOKpkVOlKlt8Lz5NxwVgWtu9FuW2pGukLtE8AWbqvY9urXAPZCQqZlOIklIjJQIqOITnuw9LEV09cgvPHXfdvNni3ldbMlIQ89zryM6dYvhYryTiEZGK4JDR3wAKJA=
+  - provider: script
+    user: "ToFuProject"
+    script: $START/anaconda_upload.sh
+    on:
+      tags: true
+      branch: master
+    skip_cleanup: true
```
Comment (lasofivec) [inline .travis.yml]: anaconda deployment being done here

File: _updateversion.py [hunk 3]
```diff
@@ -6,20 +6,21 @@
...
     # Also, when git is not available (PyPi package), use stored version.py
     version_py = os.path.join(path, 'tofu', 'version.py')
     try:
         version_git = subprocess.check_output(["git",
                                                "describe"]).rstrip().decode()
-    except:
-        with open(version_py,'r') as fh:
-            version_git = fh.read().strip().split("=")[-1].replace("'",'')
-    version_git = version_git.lower().replace('v','')
+    except subprocess.CalledProcessError:
```
Comment (lasofivec) [inline _updateversion.py]: catching right exception

File: setup.py [hunk 4]
```diff
@@ -177,15 +179,14 @@ def get_version_tofu(path=_HERE):
...
-                        "--symbolic-full-name",
                         "--abbrev-ref",
                         "HEAD",
                     ]
                 )
                 .rstrip()
                 .decode()
             )
-            if git_branch in ["master"]:
+            if git_branch in ["master", "deploy-test"]:
```
Comment (lasofivec) [inline setup.py]: updating version on both deployment branches

**Comment by Didou09** [comment]
I'll let you decide if you want to keep the branch

**Comment by lasofivec** [comment]
It's a matter of "taste" I guess. Same as with "new-doc". We can erase them, and recreate them when needed (ie. before testing a deployment and when updating the documentation). Branches are bit like "pointers" to commits, so I don't see a problem with erasing them, nor with keeping them (except for having too many active+stale branches).

**Pull request merged**